hr_course/models/hr_course.py

```diff
@@ -65,6 +76,62 @@ class HrCourse(models.Model):
         readonly=True,
     )
 
+    schedule_validity_end_date = fields.Date(
+        related="course_schedule_ids.validity_end_date"
+    )
+
+    course_attendee_ids = fields.One2many(
+        related="course_schedule_ids.course_attendee_ids"
+    )
+
+    @api.model
+    def send_course_notification_email(self):
+        company_id = self.env.context.get("company_id") or self.env.company.id
+        channel = (
+            self.env["res.company"].browse(company_id).course_expiration_channel_id
+        )
+
+        if not channel:
+            _logger.info("no channel found for course expiration alerts")
+            return False
+
+        email_template = self.env.ref(
+            "hr_course.mail_template_validity_reminder"
+        ).with_context(lang=self.env.company.partner_id.lang)
+        email_values = email_template.generate_email(self.ids, ["body_html"])
+
+        values = email_values[self.id]
+        self.with_context(
+            recipient_ids=channel.channel_last_seen_partner_ids.mapped("partner_email"),
```

Comment:
```suggestion
```

Reply:
then where to take it from for email template?

Reply:
channel_ids=channel.ids should notify correctly I suppose

Reply:
Doesn't. Running out of ideas.
![image](https://user-images.githubusercontent.com/59824990/150513114-42be036f-e857-44d4-aa9d-4d3af7e18999.png)